libs/wingsdk/src/shared-aws/function.inflight.ts

```diff
@@ -86,47 +29,80 @@ export class FunctionClient implements IFunctionClient {
       );
     }
     if (!response.Payload) {
-      return { payload: "" };
+      return "";
     }
     const value = JSON.parse(toUtf8(response.Payload)) ?? "";
+    if (typeof value !== "string") {
+      throw new Error(
+        `function returned value of type ${typeof value}, not string`
+      );
+    }
     return value;
   }
 
-  /**
-   * Invoke the function, passing the given payload as an argument.
-   *  @returns the function returned payload only
-   */
-  public async invoke(payload: string): Promise<string> {
-    const value = await this.executeFunction(payload);
-    const functionName = value?.context?.functionName;
-
-    // kind of hacky, but this is the most convenient way to pass those arguments to the calling function
-    console.log(
-      `Invoking ${functionName}:\t${Buffer.from(
-        JSON.stringify({
-          logs: value?.context?.logs,
-        }),
-        "binary"
-      ).toString("base64")}`
-    );
-
-    return this.verify(value);
-  }
-
   /**
    * Invoke the function, passing the given payload as an argument.
    *
    * @returns the function returned payload and logs
    */
   public async invokeWithLogs(payload: string): Promise<[string, Trace[]]> {
-    const traces: Trace[] = [];
+    const command = new InvokeCommand({
+      FunctionName: this.functionArn,
+      Payload: fromUtf8(JSON.stringify(payload)),
+      LogType: LogType.Tail,
```

Comment:
This is only good as a temporary solution, as it sends back only the last 4kb (1k-4k~ words) which is not a lot, also, we are losing all the logs that are not in the `test {}` scope.
I am working now on a different approach, that will involve querying logs properly (as it seems to be inevitable), that we can turn off by a flag (to overcome the extra async call)